h5grove/content.py

```diff
@@ -292,3 +292,39 @@ def get_content_from_file(
         raise create_error(404, str(e))
     finally:
         f.close()
+
+
+@contextlib.contextmanager
+def get_list_of_paths(
+    filepath: Union[str, Path],
+    base_path: Optional[str],
+    create_error: Callable[[int, str], Exception],
+    resolve_links: LinkResolution = LinkResolution.ONLY_VALID,
+    h5py_options: Dict[str, Any] = {},
+):
+    try:
+        f = h5py.File(filepath, "r", **h5py_options)
+    except OSError as e:
+        if isinstance(e, FileNotFoundError) or "No such file or directory" in str(e):
+            raise create_error(404, "File not found!")
+        if isinstance(e, PermissionError) or "Permission denied" in str(e):
+            raise create_error(403, "Cannot read file: Permission denied!")
+        raise e
```

Comment:
Duplicated a bit of `get_content_from_file`. I didn't think it was worth abstracting given this is short and only used twice

Reply:
OK for me, but risk is then that one gets modified and not the other..

Reply:
See 8a4e4f9772fcaac0dc13f2905613860444bac9cf

Reply:
:+1: